src/ui/reducers/timeline.ts

```diff
@@ -96,6 +94,19 @@ export const getUnprocessedRegions = (state: UIState) => state.timeline.unproces
 export const getRecordingDuration = (state: UIState) => state.timeline.recordingDuration;
 export const getTimelineDimensions = (state: UIState) => state.timeline.timelineDimensions;
 export const getHoveredItem = (state: UIState) => state.timeline.hoveredItem;
+export const getPaints = (state: UIState) => state.timeline.paints;
+export const getPoints = (state: UIState) => state.timeline.points;
+export const getBasicProcessingProgress = (state: UIState) => {
+  if (state.timeline.allPaintsReceived) {
+    return 1.0;
+  }
+  const maxPaint = maxBy(state.timeline.paints, p => BigInt(p.point));
+  const maxPoint = maxBy(state.timeline.points, p => BigInt(p.point));
+  if (!maxPoint || maxPoint.point === "0") {
+    return 0.0;
+  }
+  return (1.0 * (maxPaint?.time || 0)) / maxPoint.time;
+};
```

Comment:
Two thoughts:

1. This selector feels pretty heavy, given that it calls `maxBy` twice each time it's run– which loops over the entire array of paints and points. Fortunately I don't think we need to call `maxBy` since those arrays are sorted. (Can't we just look at the last element in the array?)

2. Should `getBasicProcessingProgress` be taking paints into account even if `basicProcessingLoadingBar` is disabled? (Should it not fall back to the prior behavior of comparing the latest time in the known regions?)

Reply:
1. Yes. Good point, duh! I will make that change.
2. I think either: 1) `getBasicProcessingProgress` is off, in which case this is never called, or else it is on, in which case we need paints to make any sort of reasonable estimate about how far basic processing has gotten.